model/post.go

```diff
@@ -712,18 +712,10 @@ func (o *Post) ToNilIfInvalid() *Post {
 	return o
 }
 
-func (o *Post) RemovePreviewPost() {
-	if o.Metadata == nil || o.Metadata.Embeds == nil {
-		return
-	}
-	n := 0
-	for _, embed := range o.Metadata.Embeds {
-		if embed.Type != PostEmbedPermalink {
-			o.Metadata.Embeds[n] = embed
-			n++
-		}
-	}
-	o.Metadata.Embeds = o.Metadata.Embeds[:n]
+func (o *Post) ForPlugin() *Post {
+	p := o.Clone()
+	p.Metadata = nil
```

Comment:
We use the `Metadata` field for permalink previews (a non-OG case). Is it OK to be stripping that from plugins?

Reply:
I am not (yet?) aware of the plugins needing access to the Metadata. I am somewhat concerned that in case of UpdatePost we don't seem to be rebuilding the metadata for the new Message, but that's a separate issue altogether.